blocks/components/editable/index.js

```diff
@@ -329,6 +329,7 @@ export default class Editable extends wp.element.Component {
 		} else {
 			this.editor.formatter.apply( format );
 		}
+		this.editor.setDirty( true );
```

Comment:
Apparently it isn't. I inspected the dirty state before and after the formatter call and it wasn't being set. This is probably a bug in TinyMCE though.